File: springboot-demo/src/main/java/com/itangcent/springboot/demo/cn/controller/DeprecatedUserCtrl.java
package com.itangcent.springboot.demo.cn.controller;

import com.itangcent.springboot.demo.cn.dto.IResult;
import com.itangcent.springboot.demo.cn.dto.Result;
import com.itangcent.springboot.demo.cn.model.UserInfo;
import org.springframework.web.bind.annotation.*;

import java.util.Collections;

/**
 * 用户相关(已废弃)
 *
 * @module user
 */
@Deprecated
@RestController
@RequestMapping(value = "/user/deprecated")
public class DeprecatedUserCtrl {

    /**
     * 打个招呼
     */
    @Deprecated
    @RequestMapping(value = "/index")
    public String greeting() {
        return "hello world";
    }

    /**
     * 更新用户名
     *
     * @param id      用户id
     * @param newName 新的用户名
     * @deprecated 改用{@link #update(UserInfo)}
     */
    @Deprecated
    @RequestMapping(value = "/set", method = RequestMethod.PUT)
    public Object set(Long id, String newName) {
        UserInfo userInfo = new UserInfo();
        userInfo.setId(id);
        userInfo.setName(newName);
        userInfo.setAge(45);
        return Result.success(userInfo);
    }

    /**
     * 获取用户信息
     *
     * @param id 用户id
     */
    @GetMapping("/get/{id}")
    public IResult get(@PathVariable("id") Long id) {
        UserInfo userInfo = new UserInfo();
        userInfo.setId(id);
        userInfo.setName("Tony Stark");
        userInfo.setAge(45);
        return Result.success(userInfo);
    }

    /**
     * 增加新用户
     */
    @PostMapping("/add")
    public IResult add(@RequestBody UserInfo userInfo) {
        return Result.success(userInfo);
    }

    /**
     * 更新用户信息
     */
    @PutMapping("update")
    public IResult update(@ModelAttribute UserInfo userInfo) {
        return Result.success(userInfo);
    }

    /**
     * 获取用户列表
     */
    @RequestMapping(value = "/list", method = RequestMethod.GET)
    public IResult list() {
        UserInfo userInfo = new UserInfo();
        userInfo.setId(1l);
        userInfo.setName("Tom");
        userInfo.setAge(25);
        return Result.success(Collections.singletonList(userInfo));
    }

    /**
     * 删除用户信息
     *
     * @param id 用户id
     */
    @DeleteMapping("/{id}")
    public IResult delete(@PathVariable("id") Long id) {
        UserInfo userInfo = new UserInfo();
        userInfo.setId(id);
        userInfo.setName("Tony Stark");
        userInfo.setAge(45);
        return Result.success(userInfo);
    }

}

File: cn/kotlin-demo/src/main/kotlin/com/itangcent/cn/kotlin/demo/controller/DeprecatedUserCtrl.kt
package com.itangcent.cn.kotlin.demo.controller

import com.itangcent.common.dto.IResult
import com.itangcent.common.dto.Result
import com.itangcent.common.model.UserInfo
import org.springframework.web.bind.annotation.*

import java.util.Collections

/**
 * 用户相关(已废弃)
 *
 * @module user
 */
@Deprecated("")
@RestController
@RequestMapping(value = ["/user/deprecated"])
class DeprecatedUserCtrl {

    /**
     * 打个招呼
     */
    @Deprecated("")
    @RequestMapping(value = ["/index"])
    fun greeting(): String {
        return "hello world"
    }

    /**
     *
     * 更新用户名
     *
     * @param id      用户id
     * @param newName 新的用户名
     */
    @RequestMapping(value = ["/set"], method = [RequestMethod.PUT, RequestMethod.GET])
    @Deprecated("改用{@link #update(UserInfo)}")
    operator fun set(id: Long?, newName: String): Any {
        val userInfo = UserInfo()
        userInfo.id = id
        userInfo.name = newName
        userInfo.age = 45
        return Result.success(userInfo)
    }

    /**
     * 获取用户信息
     *
     * @param id 用户id
     */
    @GetMapping("/get/{id}")
    operator fun get(@PathVariable("id") id: Long?): IResult {
        val userInfo = UserInfo()
        userInfo.id = id
        userInfo.name = "Tony Stark"
        userInfo.age = 45
        return Result.success(userInfo)
    }

    /**
     * 增加新用户
     */
    @PostMapping("/add")
    fun add(@RequestBody userInfo: UserInfo): IResult {
        return Result.success(userInfo)
    }

    /**
     * 更新用户信息
     */
    @PutMapping("update")
    fun update(@ModelAttribute userInfo: UserInfo): IResult {
        return Result.success(userInfo)
    }

    /**
     * 获取用户列表
     */
    @RequestMapping(value = ["/list"], method = [RequestMethod.GET])
    fun list(): IResult {
        val userInfo = UserInfo()
        userInfo.id = 1L
        userInfo.name = "Tom"
        userInfo.age = 25
        return Result.success(listOf(userInfo))
    }

    /**
     * 删除用户信息
     *
     * @param id 用户id
     */
    @DeleteMapping("/{id}")
    fun delete(@PathVariable("id") id: Long?): IResult {
        val userInfo = UserInfo()
        userInfo.id = id
        userInfo.name = "Tony Stark"
        userInfo.age = 45
        return Result.success(userInfo)
    }

}
